lib/protoflow-blocks/src/blocks/sys/read_socket.rs

```diff
@@ -0,0 +1,240 @@
+        }))
+    }
+}
+
+impl<T: Message + 'static> Block for ReadSocket<T> {
+    fn prepare(&mut self, _runtime: &dyn BlockRuntime) -> BlockResult {
+        let address = format!("{}:{}", &self.config.host, &self.config.port);
+        let listener = TcpListener::bind(&address)?;
+        *self.listener.lock().map_err(lock_error)? = Some(listener);
+        info!("Server listening on {}", address);
+        Ok(())
+    }
+
+    fn execute(&mut self, _: &dyn BlockRuntime) -> BlockResult {
+        let mut stream_guard = self.stream.lock().map_err(lock_error)?;
+
+        if stream_guard.is_none() {
+            let listener_lock = self.listener.lock().map_err(lock_error)?;
+            let listener = listener_lock
+                .as_ref()
+                .ok_or(BlockError::Other("Invalid TCP listener".into()))?;
+
+            let (stream, addr) = listener.accept().map_err(|e| {
+                error!("Failed to accept client connection: {}", e);
+                BlockError::Other("Failed to accept client connection".into())
+            })?;
+
+            info!("Accepted connection from {}", addr);
+            *stream_guard = Some(stream);
+        }
+
+        if let Some(stream) = stream_guard.as_mut() {
+            handle_client::<T, _>(stream, self.config.buffer_size, |message| {
+                info!("Processing received message");
+                if self.output.is_connected() {
+                    self.output.send(message)?;
+                }
+                Ok(())
+            })
+            .map_err(|e| {
+                error!("Error handling client: {}", e);
+                BlockError::Other("Error handling client".into())
+            })?;
+        }
+
+        Ok(())
+    }
+}
+
+fn lock_error<T>(err: PoisonError<T>) -> BlockError {
+    BlockError::Other(format!("Failed to acquire lock: {:?}", err))
+}
+
+fn handle_client<M, F>(
+    stream: &mut TcpStream,
+    buffer_size: usize,
+    process_fn: F,
+) -> Result<(), BlockError>
+where
+    M: Message + Default,
+    F: Fn(&M) -> PortResult<()>,
+{
+    let mut buffer = vec![0; buffer_size];
+
+    while let Ok(bytes_read) = Read::read(stream, &mut buffer) {
+        if bytes_read == 0 {
+            info!("Client disconnected");
+            break;
+        }
+
+        let message = M::decode(&buffer[..bytes_read])
+            .map_err(|_| BlockError::Other("Failed to decode message".into()))?;
+
+        info!("Received message: {:?}", message);
+
+        process_fn(&message).map_err(|e| {
+            error!("Failed to send response: {:?}", e);
+            BlockError::Other("Failed to send response".into())
+        })?;
+    }
+
+    Ok(())
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ReadSocketConfig {
+    pub host: String,
+    pub port: u16,
+    pub buffer_size: usize,
+}
+
+impl Default for ReadSocketConfig {
+    fn default() -> Self {
+        Self {
+            host: String::from("127.0.0.1"),
+            port: 7070,
```

Comment:
While localhost is a fine default for the host, it doesn't make sense to have a default port. So, cannot implement `Default` here